docs/application/web/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,445 @@
+   }
+   pipeline.registerSinkListener('sinkx', sinkListener);
+   ```
+
+After starting the pipeline, `sinkListener` will be called repeatedly.
+
+
+## Change data flow within pipeline
+You may want to set the the source of the data to one of several pipeline branches.
+It can be done with `input-selector` node and `Switch` API.
+
+We will use the pipeline from the figure below to show how to use `input-selector`, to choose between blue and green branch as the source for the white branch.
+
+![Pipeline with branches and a switch](./media/pipeline_with_switch.svg)
+
+To choose a source branch with `input-selector`:
+1. Create a pipeline:
+   ```javascript
+   var pipelineDescription = 'input-selector name=ins ! tensor_converter ! tensor_sink name=sinkx ' + // white branch
+                             'videotestsrc is-live=true ! videoconvert ! ins.sink_0 ' + // blue branch
+                             'videotestsrc num-buffers=3 is-live=true ! videoconvert ! ins.sink_1'; // green branch
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
+
+2. Get `insSwitch`:
+   ```javascript
+   var insSwitch = pipeline.getSwitch('ins');
+   ```
+
+3. You can get `Switch` pads:
+   ```javascript
+   console.log(insSwitch.getPadList()); // ["sink_0", "sink_1"]
+   ```
+
+4. Choose the pad to be used as a source for `input-selector` and the white pipeline branch:
+   ```javascript
+   insSwitch.select('sink_1'); // green branch used as the source
+   ```
+
+In the example above, we used an `input-selector` to choose from alternative sources.
+Similarily, we can use an `output-selector` to direct data to one of a set of sinks.
+
+
+## Start and stop data flow to a pipeline branch
+Use `valve` element with `Valve` API to start and stop data flow in one of a pipeline branches.
+
+To control data flow in a single pipeline branch:
+1. Create and start a pipeline with a `valve` element:
+   ```javascript
+   var pipelineDescription = 'videotestsrc is-live=true ' +
+                             '! videoconvert ' +
+                             '! videoscale ' +
+                             '! video/x-raw,format=RGBx,width=16,height=16,framerate=10/1 ' +
+                             '! tensor_converter ' +
+                             '! valve name=valve1 ' +
+                             '! fakesink';
+
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   pipeline.start();
+   ```
+
+2. Get `Valve` element from the pipeline:
+   ```javascript
+   var valve = pipeline.getValve('valve1');
+   ```
+
+3. You can check if the valve is opened or closed:
+   ```javascript
+   console.log(valve.isOpen); // true
+   ```
+
+4. Set the desired valve state:
+   ```javascript
+   valve.setOpen(false); // stop the data flow
+   ```
+
+
+## Write custom data filters
+The data flowing through the pipeline can be transformed in application, using  `CustomFilter` callbacks.
```

Comment:
Global : 1 line space between heading & next text following it 
Like this 
## Write custom data filters

The data flowing through the pipeline can be transformed in application, using  `CustomFilter` callbacks.